pandas/io/formats/latex.py

```diff
@@ -106,13 +112,60 @@ def pad_empties(x):
             raise AssertionError('column_format must be str or unicode, '
                                  'not {typ}'.format(typ=type(column_format)))
 
+        use_table_env = False
```

Comment:
I would make separate functions to handle these cases rather than big long ones.

Reply:
good suggestion.  this has been changed in the latest commit.